includes/downloads.php

```diff
@@ -23,6 +23,76 @@ function podlove_get_remote_addr() {
 	return $_SERVER['REMOTE_ADDR'];
 }
 
+function ga_track_download($request_id, $media_file, $ua_string, $ptm_context, $ptm_source) {	
+	// GA Tracking
+	$debug_ga = false;
+	$ga_collect_endpoint = 'https://www.google-analytics.com/' . ($debug_ga ? 'debug/' : '') . 'collect';
+	
+	$ga_tracking_id = trim(\Podlove\get_setting('tracking', 'ga'));
+	if (!$ga_tracking_id || $ga_tracking_id === '') {
+		return;
+	}
+	
+	$ga_params = array(
+		// Basics
+		'v' => '1', // version
+		'tid' => $ga_tracking_id, // tracking id
+		'cid' => $request_id, // client id
+		'ua' => $ua_string, // user agent override
+		'uip' => podlove_get_remote_addr(), // IP override
+		'ds' => 'podlove', // data source
+		
+		// We highjack the campaign fields for context/source data.
+		// Source / Medium maps to Podlove context / Podlove source.
+		// This way all Podlove sources can be easily grouped into GA Channels.
+		'cs' => $ptm_context, // campaign source
+		'cm' => $ptm_source, // campaign medium
+		'ci' => $media_file->episode()->number, // campaign id
```

Comment:
Performance concern: do `$episode = $media_file->episode()` once and use that instead of calling `$media_file->episode()` repeatedly as it does duplicate database requests each time.